File: app/src/main/java/com/mkrworld/libfilter/dto/Offset.java
package com.mkrworld.libfilter.dto;

public class Offset {

    private float mR;
    private float mG;
    private float mB;

    /**
     * Constructor
     *
     * @param offSet OffSet of All color
     */
    public Offset(float offSet) {
        this(offSet, offSet, offSet);
    }

    /**
     * Constructor
     *
     * @param offSetR Red-Color OffSet
     * @param offSetG Green-Color OffSet
     * @param offSetB Blue-Color OffSet
     */
    public Offset(float offSetR, float offSetG, float offSetB) {
        mR = offSetR;
        mG = offSetG;
        mB = offSetB;
    }

    /**
     * Metho dto get the Red Color Offset
     *
     * @return
     */
    public float getmR() {
        return mR;
    }

    /**
     * Metho dto get the Green Color Offset
     *
     * @return
     */
    public float getmG() {
        return mG;
    }

    /**
     * Metho dto get the Blue Color Offset
     *
     * @return
     */
    public float getmB() {
        return mB;
    }

    /**
     * Metho dto get the RGB/3 Color Offset
     *
     * @return
     */
    public float getAll() {
        return (mR + mG + mB) / 3F;
    }
}

File: app/src/main/java/com/mkrworld/libfilter/Offset.kt
package com.mkrworld.libfilter

/**
 * @author THEMKR
 */
class Offset {

    /**
     * OFFSET RED COLOR
     */
    val r: Float

    /**
     * OFFSET GREEN COLOR
     */
    val g: Float

    /**
     * OFFSET BLUE COLOR
     */
    val b: Float

    /**
     * Metho dto get the RGB/3 Color Offset
     *
     * @return
     */
    val all: Float
        get() {
            return (r + g + b) / 3f
        }

    /**
     * Constructor
     *
     * @param offSet OffSet of All color
     */
    constructor(offSet: Float) : this(offSet, offSet, offSet)

    /**
     * Constructor
     *
     * @param r Offset Red-Color
     * @param g Offset Green-Color
     * @param b Offset Blue-Color
     */
    constructor(r: Float, g: Float, b: Float) {
        this.r = r
        this.g = g
        this.b = b
    }
}